src/redux/actions/index.js

```diff
@@ -22,10 +22,19 @@ export const exitPreview = () => ({
 
 export const generateMapPins = () => {
   return function(dispatch){
-    fetch('api/locations')
+    fetch('http://45.55.2.200/api/location', {
+      method: 'GET',
+      headers: new Headers({
+        'Content-Type': 'application/json',
+      })
+    })
     .then((mapData) => {
-      dispatch({ type: 'GOT_ALL_LOCATION_DATA', payload: mapData })
+      // console.log('worked')
+      mapData.json().then(mapData => {
+        console.log(mapData)
+        dispatch({ type: 'GOT_ALL_LOCATION_DATA', payload: mapData })
+      })
     })
```

Comment:
The fetch code here is not wrong. However, you could have written it like this: 
```js

.then(response => response.json())
.then(mapData => dispatch({ type: 'GOT_ALL_LOCATION_DATA', payload: mapData })
```

This is more concise (fewer brackets). More importantly, returning the `.json()` from the first then, allows you to chain another `.then` from the `fetch` call, instead of directly off of the `.json()` call. This is how promises work, if the result of a promise is another promise, you can chain `.then` calls. Have a close look at this code and you should be able to spot this behavior.